lib/posix/pthread.c

```diff
@@ -186,12 +187,11 @@ int pthread_create(pthread_t *newthread, const pthread_attr_t *attr,
 	 * cannot fail.
 	 */
 	(void)pthread_mutex_init(&thread->state_lock, NULL);
-	(void)pthread_mutex_init(&thread->cancel_lock, NULL);
```

Comment:
I assume the spinlock doesn't need to be initialized because 0 means unlocked?

Reply:
Spinlocks don't need to be initialized (at all).